src/pages/IndexPage.vue

```diff
@@ -1,10 +1,87 @@
-<script setup>
-import { useFunctionsStore } from 'stores/functions.js'
-</script>
-
 <template>
   <q-page class="flex flex-center">
-    <img alt="Quasar logo" src="~assets/quasar-logo-vertical.svg" style="width: 200px; height: 200px" />
-    <q-btn @click="useFunctionsStore().test()">Test</q-btn>
+    <div id="map" :style="{ height: '100vh', width: '100%' }"></div>
   </q-page>
 </template>
+
+<script>
+import mapboxgl from 'mapbox-gl'
+
+export default {
+  // mounted() {
+  //   mapboxgl.accessToken =
+  //     'pk.eyJ1IjoiYWJkdWxsYWh6YWhpZDEwIiwiYSI6ImNsZjJ5M3UybzBuYWczc256MXdpMGZkMGsifQ.sUQa5jZuO5tn1Nnbjw6CLw'
+  //   const map = new mapboxgl.Map({
+  //     container: 'map',
+  //     style: 'mapbox://styles/mapbox/streets-v11',
+  //     center: [-0.5608289, 51.2426316], // University of Surrey
+  //     zoom: 12
+  //   })
+
+  //   // Add the missing image to the map
+  //   map.loadImage('https://docs.mapbox.com/mapbox-gl-js/assets/custom_marker.png', function (error, image) {
+  //     if (error) throw error
+  //     map.addImage('rectangle-yellow-6', image)
+  //     // Create the marker using the new image
+  //     const marker = new mapboxgl.Marker({
+  //       color: 'yellow',
+  //       scale: 0.5,
+  //       // Use the new image for the marker
+  //       icon: 'rectangle-yellow-6'
+  //     })
+  //       .setLngLat([-0.5608289, 51.2426316])
+  //       .addTo(map)
+  //   })
+  // }
+  mounted() {
+    mapboxgl.accessToken =
+      'pk.eyJ1IjoiYWJkdWxsYWh6YWhpZDEwIiwiYSI6ImNsZjJ5M3UybzBuYWczc256MXdpMGZkMGsifQ.sUQa5jZuO5tn1Nnbjw6CLw'
```

Comment:
Acknowledged; no changes have been made on this yet. @endrizzimarco @georgi-ivanov19 if you can provide env file I will make changes

Reply:
put it on whatsapp. For reference on how to use it: https://vitejs.dev/guide/env-and-mode.html

Reply:
i did make a .env file and added my api key, but map isnt loading so i wrote the code but commented it out

Reply:
If you commit and push we will be able to see you changes and try to help you. Make sure to commit and push before adding a comment, otherwise there is not much we can do

Reply:
Having issues integrating the file. I created the .env file and created a variable (you should see it in .env(last one)). I initially called package.env.VITE_MAP_API_KEY but the map was not loading

Reply:
yes commited now @georgi-ivanov19